src/NHibernate/Persister/Collection/AbstractCollectionPersister.cs

```diff
@@ -1503,10 +1503,16 @@ protected virtual string FilterFragment(string alias)
 
 		public virtual string FilterFragment(string alias, IDictionary<string, IFilter> enabledFilters)
 		{
+			var filterFragment = FilterFragment(alias);
+			if (!filterHelper.IsAffectedBy(enabledFilters))
+				return filterFragment;
+
+			if (ElementType.IsEntityType && elementPersister is AbstractEntityPersister ep)
+				return ep.FilterFragment(filterHelper, alias, enabledFilters, filterFragment);
```

Comment:
nit: is it possible to remove up-casting here?

Reply:
Yes by adding `FilterFragment` to `IEntityPersister` in 6.0. I've added comment about it: 
https://github.com/bahusoid/nhibernate-core/blob/d49a269cb58b30f24d18e765dd715326febcb457/src/NHibernate/Persister/Entity/AbstractEntityPersister.cs#L3735